src/uraft/saunafs-uraft-helper.in

```diff
@@ -151,13 +151,47 @@ saunafs_metadata_version() {
 	esac
 }
 
+have_master_role() {
+	local local_status="$(saunafs_admin metadataserver-status --porcelain "${matocl_host}" "${matocl_port}")"
+	[ "$(awk '{print $1}' <<< "${local_status}" 2>/dev/null)" == "master" ]
+}
+
+is_ip_present() {
+	local ip="${1}"
+	local iface="${2:-}"
+	local -a ip_command_extra_args=()
+	if [ -n "${iface}" ]; then
+		ip_command_extra_args+=("dev" "${iface}")
+	fi
+	ip addr show "${ip_command_extra_args[@]}" | grep -wq -m1 "${ip}"
+}
+
+is_floating_ip_present() {
+	if is_ip_present "${ipaddr}" "${iface}"; then
+		return 0
+	fi
+	if [ -n "${ipaddr2}" ] && is_ip_present "${ipaddr2}" "${iface2}"; then
+		return 0
+	fi
+	return 1
+}
+
 saunafs_isalive() {
-	saunafs_master isalive
-	if [[ $? == 0 ]] ; then
-		echo -n alive
-	else
-		echo -n dead
+	# First, check if Saunafs master reports being alive
+	if ! saunafs_master isalive; then
+		echo -n 'dead'
+		return
 	fi
+
+	load_config
```

Comment:
Why is needed or how does it help `load_config` function in `saunafs_isalive`?
Previous version did not use it.

Reply:
uRaft is also very time sensitive, I'm not sure what exactly happens when isalive doesn't report back in time (there is probably a timeout somewhere). `load_config` could take a while and may cause timeouts

Reply:
I would recommend passing down the ipaddr, iface, etc. from the C++ code by setting the environment variables after reading the config in the main uraft program.